core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java

```diff
@@ -84,6 +88,9 @@ private void addReflectiveHierarchy(ReflectiveHierarchyBuildItem i, Type type, S
         } else if (type instanceof ParameterizedType) {
             ParameterizedType p = (ParameterizedType) type;
             addReflectiveHierarchy(i, p.owner(), processedReflectiveHierarchies, unindexedClasses);
+            if (!IGNORED_PARAMETERIZED_TYPE_PACKAGE_NAMES.contains(p.name().toString())) {
+                addClassTypeHierarchy(i, p.name(), processedReflectiveHierarchies, unindexedClasses);
```

Comment:
I think that `p.owner()` was supposed to do what you just added. So if it doesn't do it, we can drop it.

Reply:
That's what I thought as well but wanted to check. I'll update the PR later on